frontend/src/pages/admin/index.svelte

```diff
@@ -0,0 +1,404 @@
+<script>
+	import {metatags} from '@roxi/routify';
+	import {onMount} from 'svelte';
+	import {
+		StudentSearch,
+		RoleSelect,
+		ToastContainer,
+		Form,
+		Input,
+		Loading,
+	} from '../../components';
+	let toastContainer;
+
+	let selectedStudent = null;
+	let studentSearch;
+	let roleSelect;
+	let submitStatus = null;
+	let resetTimeout;
+
+	let formRefreshControl = {};
+
+	metatags.title = 'Admin Home - Kitcoin';
+
+	async function manageUser(e) {
+		e.preventDefault();
+		if (!manageFormData.isValid) return false;
+		submitStatus = 'LOADING';
+		let res = await fetch(
+			selectedStudent
+				? `/api/users/${selectedStudent.value}`
+				: `/api/users`,
+			{
+				method: selectedStudent ? 'PATCH' : 'POST',
+				headers: {
+					'Content-Type': 'application/json',
+				},
+				body: JSON.stringify({
+					name: manageFormData.values.name,
+					googleID: manageFormData.values.googleID,
+					email: manageFormData.values.email || null,
+					schoolID: manageFormData.values.schoolID || null,
+					balance: parseFloat(manageFormData.values.balance),
+					balanceExpires: manageFormData.values.balanceExpires
+						? new Date(
+								manageFormData.values.balanceExpires,
+						  ).toISOString()
+						: null,
+					weeklyBalanceMultiplier: manageFormData.values
+						.weeklyBalanceMultiplier
+						? parseFloat(
+								manageFormData.values.weeklyBalanceMultiplier,
+						  )
+						: null,
+					roles: manageFormData.values.roles.map(x => x.value),
+				}),
+			},
+		).catch(() => null);
+
+		if (res && res.ok) {
+			submitStatus = 'SUCCESS';
+			toastContainer.toast(
+				selectedStudent ? 'User updated' : 'User created',
+				'success',
+			);
+			setTimeout(() => {
+				submitStatus = null;
+			}, 3000);
+
+			try {
+				let json = await res.json();
+				studentSearch.value = {text: json.name, value: json._id};
+				studentSearch.query = json.name;
+				setData(json);
+			} catch (e) {}
+		} else {
+			clearTimeout(resetTimeout);
+			submitStatus = 'ERROR';
+			toastContainer.toast(
+				`Failed to ${selectedStudent ? 'update' : 'create'} user`,
+				'error',
+			);
+			resetTimeout = setTimeout(() => {
+				submitStatus = null;
+			}, 3000);
+		}
+
+		return false;
+	}
+
+	async function deleteUser() {
+		if (
+			!confirm(`Are you sure you want to delete ${selectedStudent.text}?`)
+		)
+			return;
+		submitStatus = 'LOADING';
+
+		let res = await fetch(`/api/users/${selectedStudent.value}`, {
+			method: 'DELETE',
+		}).catch(() => null);
+
+		submitStatus = res && res.ok ? 'SUCCESS' : 'ERROR';
+		if (submitStatus == 'SUCCESS') {
+			toastContainer.toast(`${selectedStudent.text} deleted`, 'success');
+			studentSearch.el.setValue(null, null);
+			setData();
+		} else {
+			toastContainer.toast('Error deleting item.', 'error');
```

Comment:
Shouldn't this say "Error deleting user." rather than "Error deleting item."

Reply:
Copy paste moment